integration-test-chronicle/src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/chronicle/ITVerifyChronicleBackend.java

```diff
@@ -0,0 +1,231 @@
+            superSession.execute(new SimpleStatement("DROP ROLE IF EXISTS " + SUITE_TEST_USER_PREFIX + i));
+        }
+        superSession.close();
+        superCluster.close();
+
+        try (Cluster cassandraCluster = cdt.createCluster();
+             Session cassandraSession = cassandraCluster.connect())
+        {
+            cassandraSession.execute(new SimpleStatement("DROP ROLE IF EXISTS " + SUITE_SUPER_USER));
+        }
+    }
+
+    @Test
+    public void simpleUpdateIsLogged()
+    {
+        givenTable("ks1", "tbl");
+        String username = givenSuperuserWithMinimalWhitelist();
+
+        testSession.execute("UPDATE ks1.tbl SET value = 'hepp' WHERE key = 88");
+
+        thenChronicleLogContainEntryForUser("UPDATE ks1.tbl SET value = 'hepp' WHERE key = 88", username);
+    }
+
+    @Test
+    public void preparedInsertIsLogged()
+    {
+        givenTable("ks2", "tbl");
+        String username = givenSuperuserWithMinimalWhitelist();
+
+        PreparedStatement preparedStatement = testSession.prepare("INSERT INTO ks2.tbl (key, value) VALUES (?, ?)");
+        testSession.execute(preparedStatement.bind(5, "hepp"));
+
+        thenChronicleLogContainEntryForUser("INSERT INTO ks2.tbl (key, value) VALUES (?, ?)[5, 'hepp']", username);
+    }
+
+    private void thenChronicleLogContainEntryForUser(String operation, String username)
+    {
+        List<AuditRecord> records = getRecords();
+        assertThat(records).hasSize(1);
+
+        AuditRecord record = records.get(0);
+        assertThat(record.getOperation()).isEqualTo(operation);
+        assertThat(record.getUser()).isEqualTo(username);
+        assertThat(record.getStatus()).isEqualTo("ATTEMPT");
+        assertThat(record.getClient()).isEqualTo(InetAddress.getLoopbackAddress());
+        assertThat(record.getTimestamp()).isLessThan(System.currentTimeMillis());
+    }
+
+    private List<AuditRecord> getRecords()
+    {
+        List<AuditRecord> records = new ArrayList<>();
+
+        while (reader.hasRecordAvailable())
+        {
+            AuditRecord record = reader.nextRecord();
+            records.add(record);
+        }
+
+        return records;
+    }
+
+    private void givenKeyspace(String keyspace)
+    {
+        superSession.execute(new SimpleStatement(
+        "CREATE KEYSPACE IF NOT EXISTS " + keyspace + " WITH REPLICATION = {'class' : 'SimpleStrategy', 'replication_factor' : 1} AND DURABLE_WRITES = false"));
+    }
+
+    private void givenTable(String keyspace, String table)
+    {
+        givenKeyspace(keyspace);
+        superSession.execute(new SimpleStatement(
+        "CREATE TABLE IF NOT EXISTS " + keyspace + "." + table + " (key int PRIMARY KEY, value text)"));
+    }
+
+    private static String givenSuperuserWithMinimalWhitelist()
+    {
+        return testUsername;
+    }
+
+    private static String givenUniqueSuperuserWithMinimalWhitelist()
+    {
+        String username = SUITE_TEST_USER_PREFIX + usernameNumber.getAndIncrement();
```

Comment:
Why is an AtomitInteger needed here? This method is only called once (from the beforeClass method)